pkg/didcomm/protocol/didexchange/service.go

```diff
@@ -39,15 +64,39 @@ type provider interface {
 type Service struct {
 	outboundTransport transport.OutboundTransport
 	store             storage.Store
+	callbackChannel   chan didCommChMessage
+	actionEvent       chan<- event.DIDCommEvent
+	streamEvents      []chan<- dispatcher.DIDCommMsg
+	lock              sync.RWMutex
+	streamLock        sync.RWMutex
+	execute           bool
 }
 
 // New return didexchange service
 func New(store storage.Store, prov provider) *Service {
-	return &Service{outboundTransport: prov.OutboundTransport(), store: store}
+	svc := &Service{
+		// TODO Outbound dispatcher - https://github.com/hyperledger/aries-framework-go/issues/259
+		outboundTransport: prov.OutboundTransport(),
+		store:             store,
+		// TODO channel size - https://github.com/hyperledger/aries-framework-go/issues/246
+		callbackChannel: make(chan didCommChMessage, 10),
+		// set execute to false. Consumers have to enable this by setting either RegisterActionEvent() or
+		// RegisterAutoExecute()
+		execute: false,
+	}
+
+	svc.startInternalListener()
+
+	return svc
 }
 
 // Handle didexchange msg
 func (s *Service) Handle(msg dispatcher.DIDCommMsg) error {
+	// throw error if there are no action events are registered or auto execute set (if it's not outbound)
+	if !msg.Outbound && !s.execute {
```

Comment:
This is not checking whether "action events" have been registered. If there are, then the msg should be processed, even if auto-execute is disabled.

Reply:
s.execute is derived based on actionEvent or AutoExecute registration.